sklearn/model_selection/_search.py

```diff
@@ -551,41 +552,78 @@ def _fit(self, X, y, labels, parameter_iterable):
 
         candidate_params = parameters[::n_splits]
         n_candidates = len(candidate_params)
 
+        # if one choose to return train score, reshape the train_scores array
+        if self.return_train_score:
+            train_scores = np.array(train_scores,
+                                    dtype=np.float64).reshape(n_candidates,
+                                                              n_splits)
         test_scores = np.array(test_scores,
                                dtype=np.float64).reshape(n_candidates,
                                                          n_splits)
         # NOTE test_sample counts (weights) remain the same for all candidates
         test_sample_counts = np.array(test_sample_counts[:n_splits],
                                       dtype=np.int)
 
-        # Computed the (weighted) mean and std for all the candidates
+        # Computed the (weighted) mean and std for test scores
         weights = test_sample_counts if self.iid else None
-        means = np.average(test_scores, axis=1, weights=weights)
-        stds = np.sqrt(np.average((test_scores - means[:, np.newaxis]) ** 2,
-                                  axis=1, weights=weights))
+        test_means = np.average(test_scores, axis=1, weights=weights)
+        test_stds = np.sqrt(
+            np.average((test_scores - test_means[:, np.newaxis]) ** 2, axis=1,
+                       weights=weights))
+
+        time = np.array(time, dtype=np.float64).reshape(n_candidates, n_splits)
+        time_means = np.average(time, axis=1)
+        time_stds = np.sqrt(np.average((time - time_means[:, np.newaxis]) ** 2,
+                                       axis=1))
 
         cv_results = dict()
+
+        cv_results["mean_time"] = time_means
+        cv_results["std_time"] = time_stds
+
         for split_i in range(n_splits):
             cv_results["split%d_test_score" % split_i] = test_scores[:,
                                                                      split_i]
-        cv_results["mean_test_score"] = means
-        cv_results["std_test_score"] = stds
-
-        ranks = np.asarray(rankdata(-means, method='min'), dtype=np.int32)
-
-        best_index = np.flatnonzero(ranks == 1)[0]
+        cv_results["mean_test_score"] = test_means
+        cv_results["std_test_score"] = test_stds
+
+        if self.return_train_score:
+            train_means = np.average(train_scores, axis=1)
+            train_stds = np.sqrt(
+                np.average((train_scores - train_means[:, np.newaxis]) ** 2,
+                           axis=1))
+            for split_i in range(n_splits):
+                cv_results["split%d_train_score" % split_i] = (
+                    train_scores[:, split_i])
+            cv_results["mean_train_score"] = train_means
+            cv_results["std_train_score"] = train_stds
+            cv_results["rank_train_score"] = np.asarray(rankdata(-train_means,
```

Comment:
I don't generally think ranking by training score would be useful / well advised. I'd happily be wrong if someone stated otherwise.